src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -548,8 +550,7 @@ vector<UInt> SpatialPooler::initMapPotential_(UInt column, bool wrapAround) {
 
   const UInt numPotential = (UInt)round(columnInputs.size() * potentialPct_);
   const auto selectedInputs = rng_.sample<UInt>(columnInputs, numPotential);
-  const vector<UInt> potential = VectorHelpers::sparseToBinary<UInt>(selectedInputs, numInputs_);
-  return potential;
+  potential.setSparse(selectedInputs);
```

Comment:
This is not sorted!  This is the bug causing the `exactOutput` test to fail.

Because the potential pool is not sorted, the initial Permanences are randomly generated differently than before.

This also fails in debug mode where there is an `ASSERT( is_sorted )` for this type of error.